Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs

```diff
@@ -351,28 +358,41 @@ public static List<TreeViewItemData<ActionOrBindingData>> GetActionsAsTreeViewDa
             return actionItems;
         }
 
-        private static string GetHumanReadableBindingName(SerializedInputBinding serializedInputBinding, int currentControlSchemeIndex, InputControlScheme? currentControlScheme)
+        private static string GetHumanReadableBindingName(SerializedInputBinding serializedInputBinding, InputControlScheme? currentControlScheme, SerializedProperty allControlSchemes)
         {
             var name = InputControlPath.ToHumanReadableString(serializedInputBinding.path);
             if (String.IsNullOrEmpty(name))
                 name = "<No Binding>";
-            if (!IsBindingPartOfCurrentControlScheme(serializedInputBinding, currentControlScheme, currentControlSchemeIndex))
+            if (IsBindingAssignedToNoOrAllControlSchemes(serializedInputBinding, allControlSchemes, currentControlScheme))
                 name += " {GLOBAL}";
             return name;
         }
 
-        private static bool IsBindingPartOfCurrentControlScheme(SerializedInputBinding serializedInputBinding, InputControlScheme? currentControlScheme, int currentControlSchemeIndex)
+        private static bool IsBindingAssignedToNoOrAllControlSchemes(SerializedInputBinding serializedInputBinding, SerializedProperty allControlSchemes, InputControlScheme? currentControlScheme)
         {
-            if (currentControlScheme.HasValue && currentControlSchemeIndex >= 0)
+            if (allControlSchemes.arraySize <= 0 || !currentControlScheme.HasValue || string.IsNullOrEmpty(currentControlScheme.Value.name))
+                return false;
+            if (serializedInputBinding.controlSchemes.Length <= 0)
+                return true;
+            return false;
+        }
+
+        private static bool IsBindingPartOfCurrentControlScheme(SerializedInputBinding serializedInputBinding, InputControlScheme? currentControlScheme)
+        {
+            if (currentControlScheme.HasValue && !string.IsNullOrEmpty(currentControlScheme.Value.name))
+            {
+                if (serializedInputBinding.controlSchemes.Length <= 0)
+                    return true;
                 return serializedInputBinding.controlSchemes.Contains(currentControlScheme.Value.name);
+            }
 
             return true;
```

Comment:
Should this be `return false`?

Reply:
I updated the method name to be more descriptive & added comments @jamesmcgill